### DISCUSSION TITLE
Airflow dag broken

### DISCUSSION THREAD

**Discussion by Praful19** [apache/airflow#30275] [Q&A]
we are doing deployment using jenkins we have installed module in file called requirement.txt. when we did deployment we are see below error.

airflow dag "broken" no module named sendgrid. 

we have apache-airflow-providers-sendgrid in the requirement file but still we are facing the issue. not sure what is the issue.

#dagbroken #sendgrid

**Comment by kobethuwis** [comment]
Could you share the import statement inside the DAG and some details about your way of deploying? You could always take a look at [Module Management](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/modules_management.html) or [Plugins](https://airflow.apache.org/docs/apache-airflow/2.5.1/authoring-and-scheduling/plugins.html) when you're having troubles with Python packages in DAGs.

**Reply by potiuk** [reply]
Well. You really have to look at the way how your requirements.txt are working and whether they install things proeperly. 

If you run it on your own - you need to see what is getting installed and see if in the target environment of yours the right packages are installed. 

If you are running it in managed version of Airlfow (for example AWS)- then you need to understand how requirements.txt work and how they are installed there - but this is really question of the support team of the managed services.

In either case this is a question on how your deployment pipline works, not something people here without knowing your pipeline will be able to help with.

**Reply by Praful19** [reply]
We are using managed version of Airflow from AWS. we have dags file and ansible to do certain task like creating folder and S3 sync. this is done using jenkins pipeline. from the pipeline we see that dags are updated to S3. The Managed Airflow MWAA calls this requirment.txt into install certain modules. however we are seeing issue only with sendgrid module. all the dags not able recognize sendgrid module. we checked cloudwatch logs however, we didn't find any. 

One recent change is we did the deployment using jenkins which recently used EC2 on demand instance previously it was using on-prem. also we upgraded python version in the ansible from python 2.7 to python 3. I hope this isn't the cause of the issue. because we don't see anywhere in the build the python upgrade is cause of this.

any suggestion/solution please let me know if you need some more details. Thanks

**Reply by potiuk** [reply]
I think you need to check if ansible is installed after you added it. This is really a question to MWAA support - with understanding on how this are installed and which packages you use and how yoru python 2 vs. python 3 is used. It's a very specific question to your packages and dependencies and has not much to do with Airflow.
